shell/platform/android/io/flutter/embedding/android/MotionEventTracker.java

```diff
@@ -0,0 +1,83 @@
+package io.flutter.embedding.android;
+
+import android.view.MotionEvent;
+import androidx.annotation.Nullable;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicLong;
+
+/** Tracks the motion events received by the FlutterView. */
+public final class MotionEventTracker {
+
+  /** Represents a unique identifier corresponding to a motion event. */
+  public static class MotionEventId {
+    private static final AtomicLong ID_COUNTER = new AtomicLong(0);
+    private final long id;
+
+    private MotionEventId(long id) {
```

Comment:
would this wrapper object provide something that just the return value of `AtomicLong` doesn't?

Reply:
Got it. I don't feel strongly about it, so your call!